openbsd/libs.md

```diff
@@ -0,0 +1,128 @@
+# Instructions for obtaining EFI related libraries on OpenBSD
+
+Read [README.md](./README.md) first for general information. Note that building
+ports requires system with installed X11 parts of base system, you can load
+`bsd.rd` from `boot >` prompt, pick "Update" and install the missing packages if
+needed.
+
+The plan is:
+ * install kernel with support for EFI variables
+ * prepare environment for building ports
+ * build and install `efivar` port
+ * compile and install test tool that uses `libefivar`
+ * verify that it works
+
+## Install kernel with support for EFI variables
+
+Follow steps outlined in [efivars.md](./efivars.md) for this.
+
+## Preparing for using ports
+
+To prepare using ports as a regular user `user` perform these steps as `root`:
+
+```
+cat > /etc/mk.conf <<EOF
+WRKOBJDIR=/usr/obj/ports
+DISTDIR=/usr/distfiles
+PACKAGE_REPOSITORY=/usr/packages
+EOF
+
+mkdir -p /usr/ports /usr/distfiles /usr/packages
+chown user:wobj /usr/distfiles /usr/packages/
+
+# to be able to use ports as a regular user which is part of `wheel` group
+chmod -R g+w /usr/ports/
+```
+
+## Building and installing the port
+
+As a regular user assuming `doas` is configured:
+
+```
+git clone --depth 1 -b efivar https://github.com/3mdeb/ports /usr/ports
+
+cd /usr/ports/devel/efivar
+# update "distinfo" file in case it diverged due to active development
+make makesum
+# build the port as a regular user
+make package
+# install as root
+doas make install
+```
+
+## Prepare user-space tool
+
+```
+# build sample tool
+make -C /usr/src/usr.sbin/efivarlibtest
```

Comment:
@SergiiDmytruk Missing header? Any idea?

```
efivarlibtest.c:9:10: fatal error: 'efivar/efivar.h' file not found
#include <efivar/efivar.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
Have you installed the port?

Reply:
Yes I have. I could use the efivar binary. I may give it another try if you are certain it should work.

Reply:
I see `-I/usr/local/include` in https://github.com/3mdeb/openbsd-src/commit/0eec59058955b68642aefbe461aa249ae96621f6 and port should install `/usr/local/include/efivar/efivar.h`. You can check if headers are there and contents of `/usr/src/usr.sbin/efivarlibtest/Makefile`.

Reply:
@SergiiDmytruk I have finally suceeded with following notes:

1. I needed to run `doas susypgrade` and upgrade the OpenBSD version to 7.2 first, as the `pkg_add` was failing to install any packages - as here: https://www.reddit.com/r/openbsd/comments/v4t87e/cant_update_packages_anymore/

2. When running `make package` with user priviledges, it failed when trying to install some packages via `pkg_add` - I'm not sure what is the recommendation there:

```
===>  Installing dwz-0.14 from /usr/packages/amd64/all/
pkg_add: pkg_add must be run as root
```

3. Because of the point 2, I installed the dependencies manually first, before running `make package`. These were: `libiconv gmake dwz lunzip`

Reply:
Please add some notes on how to handle such cases and we can close that MR from my perspective.

Reply:
I know that error, but I must have installed dependencies before switching the kernel, so `make package` didn't error for me. `pkg_add` seems to use kernel version to know which ports to pull, which is why you had to upgrade or load system with the original kernel (and original bootloader) to work around incompatibility.